openpathsampling/analysis/tis/flux.py

```diff
@@ -0,0 +1,330 @@
+
+class MinusMoveFlux(MultiEnsembleSamplingAnalyzer):
+    """
+    Calculating the flux from the minus move.
+
+    Raises
+    ------
+    ValueError
+        if the number of interface sets per minus move is greater than one.
+        Cannot use Minus Move flux calculation with multiple interface set
+        TIS.
+
+    Parameters
+    ----------
+    scheme: :class:`.MoveScheme`
+        move scheme that was used (includes information on the minus movers
+        and on the network)
+    flux_pairs: list of 2-tuple of :class:`.Volume`
+        pairs of (state, interface) for calculating the flux out of the
+        volume and through the state. Default is `None`, in which case the
+        state and innermost interface are used.
+    """
+    def __init__(self, scheme, flux_pairs=None):
+        super(MinusMoveFlux, self).__init__()
+        # error string we'll re-use in a few places
+        mistis_err_str = ("Cannot use minus move flux with multiple "
+                          + "interface sets. ")
+        self.scheme = scheme
+        self.network = scheme.network
+        self.minus_movers = scheme.movers['minus']
+        for mover in self.minus_movers:
+            n_innermost = len(mover.innermost_ensembles)
+            if n_innermost != 1:
+                raise ValueError(
+                    mistis_err_str + "Mover " + str(mover) + " does not "
+                    + "have exactly one innermost ensemble. Found "
+                    + str(len(mover.innermost_ensembles)) + ")."
+                )
+
+        if flux_pairs is None:
+            # get flux_pairs from network
+            flux_pairs = []
+            minus_ens_to_trans = self.network.special_ensembles['minus']
+            for minus_ens in self.network.minus_ensembles:
+                n_trans = len(minus_ens_to_trans[minus_ens])
+                if n_trans > 1:  # pragma: no cover
+                    # Should have been caught be the previous ValueError. If
+                    # you hit this, something unexpected happened.
+                    raise ValueError(mistis_err_str + "Ensemble "
+                                     + repr(minus_ens) + " connects "
+                                     + str(n_trans) + " transitions.")
+                trans = minus_ens_to_trans[minus_ens][0]
+                innermost = trans.interfaces[0]
+                state = trans.stateA
+                # a couple assertions as a sanity check
+                assert minus_ens.state_vol == state
+                assert minus_ens.innermost_vol == innermost
+                flux_pairs.append((state, innermost))
+
+        self.flux_pairs = flux_pairs
+
+    def _get_minus_steps(self, steps):
+        """
+        Selects steps that used this object's minus movers
+        """
+        return [s for s in steps
+                if s.change.canonical.mover in self.minus_movers
+                and s.change.accepted]
+
+    def trajectory_transition_flux_dict(self, minus_steps):
```

Comment:
The flux_dict looks like a complicated object. Is there a reason not to create a class for it? I mean, is a user meant to create this object by himself for some reason? I just thought we could hide some of the complexity this way.